VectorsCPP/Vector.hpp

```diff
@@ -0,0 +1,167 @@
+#pragma once
+#include <iostream>
+#include <initializer_list>
+#include <algorithm>
+
+
+class Vector
+{
+public:
+    // Constructors
+    Vector()
+        :size(0),
+        capacity(5),
+        array(new int[capacity])
+    {
+
+    }
+
+    Vector(const Vector& rhs)
+        :size(rhs.size),
+        capacity(rhs.capacity + 5),
+        array(new int[capacity])
+    {
+        for (int i = 0; i < rhs.Size(); i++)
+        {
+            array[i] = rhs.array[i];
+        }
+    }
+
+    Vector(int elements, int value)
+        : size(elements),
+        capacity(elements + 5),
+        array(new int[capacity])
+    {
+        for (int i = 0; i < size; i++)
+        {
+            array[i] = value;
+        }
+    }
+
+    Vector(const std::initializer_list<int>& list)
+        : size(0),
+        capacity(list.size() + 5),
+        array(new int[capacity])
+    {
+        for (int i : list)
+        {
+            PushBack(i);
+        }
+    }
+
+    // Destructor
+    ~Vector()
+    {
+        delete[] array;
+    }
+
+    // Functions
+    void PushBack(int value)
+    {
+        if (size < capacity)
+        {
+            array[size] = value;
+            ++size;
+        }
+        else
+        {
+            capacity *= 2;
+            int* newarray = new int[capacity];
+            for (int i = 0; i < size; i++)
+            {
+                newarray[i] = array[i];
+            }
+            newarray[size] = value;
+            ++size;
+            delete[] array;
+            array = newarray;
+        }
+    }
+    //remove the last number from the vector
+    void PopBack()
+    {
+        if (size > 0)
+        {
+            --size;
+        }
+    }
+    //clears the vector and make the size equal to 0
+    void Clear()
+    {
+        size = 0;
+    }
+
+    //using a built in librerty to reverse the numbers
+    void Reverse()
+    {
+        std::reverse(array, array + size);
+    }
+
+    //Displays the Vector
+    void Display() const
```

Comment:
why is this function necessary if you have the `<<` operator

Reply:
Its the same code from before. you need to look at "v4 final"

Reply:
it's still here in the updated version 😜